File: DSA-DAILY/29-07-2025/2411. Smallest Subarrays With Maximum Bitwise OR@leetcode-med.java
class Solution {
    public int[] smallestSubarrays(int[] nums) {
        int n=nums.length;
        int[] ans=new int[n];
        for(int i=0;i<n;i++){
            int x=nums[i];
            ans[i]=1;
            for(int j=i-1;j>=0&&(nums[j]|x)!=nums[j];--j){
                nums[j]|=x;
                ans[j]=i-j+1;
            }
        }
        return ans;
    }
    public static void main(String[] args) {
        Solution solution = new Solution();
        int[] nums = {1, 2, 3, 4};
        int[] result = solution.smallestSubarrays(nums);
        for (int res : result) {
            System.out.print(res + " ");
        }
        // Output: [4, 3, 2, 1]
    }
}

File: 29-07-2025/2411. Smallest Subarrays With Maximum Bitwise OR@leetcode-med.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var smallestSubarrays = function(nums) {
    const n = nums.length;
    const ans = new Array(n).fill(1);

    for (let i = 0; i < n; i++) {
        let x = nums[i];
        for (let j = i - 1; j >= 0 && (nums[j] | x) !== nums[j]; j--) {
            nums[j] |= x;
            ans[j] = i - j + 1;
        }
    }

    return ans;
};

// Node.js input handling
const readline = require('readline');
const rl = readline.createInterface({
    input: process.stdin,   
    output: process.stdout
});

rl.question('Enter array elements separated by space: ', (input) => {
    const nums = input.trim().split(/\s+/).map(Number);
    const result = smallestSubarrays(nums);
    console.log('Smallest subarrays with maximum bitwise OR:', result);
    rl.close();
});
